src/content/app/entity-viewer/state/gene-view/transcripts/geneViewTranscriptsSlice.ts

```diff
@@ -275,67 +292,76 @@ type UpdateFilterPanelPayload = {
   filterPanelOpen: boolean;
 };
 
+export const restoreTranscriptsFiltersAndSorting =
+  (): ThunkAction<void, any, null, Action<string>> => (dispatch) => {
+    const storedFiltersAndSortingData =
+      entityViewerStorageService.getGeneViewTranscriptsState();
+
+    if (!storedFiltersAndSortingData) {
+      return;
+    }
+    dispatch(
+      transcriptsSlice.actions.loadInitialState(storedFiltersAndSortingData)
+    );
+  };
```

Comment:
Could you use the `createAsyncThunk` function from redux toolkit to create this thunk? Then you'll be able to use it in the `extraReducers` object inside the `createSlice`, its return value being the action payload that an extra reducer receives (see `entityViewerBookmarksSlice.ts` for comparison). This will allow you to remove the `loadInitialState` reducer.

By the way, it seems that this function is going to load everything that's stored about this slice, not just filters and sorting rules. Let's remove these words from the name of the function. `restoreGeneViewTranscriptsSlice` perhaps? Or `loadPreviouslyViewedTranscripts`? Or something.